Tasks/AzureRmWebAppDeployment/task.json

```diff
@@ -38,23 +38,54 @@
             "name": "PostDeploymentAction",
             "displayName": "Post Deployment Action",
             "isExpanded": false,
-            "visibleRule": "WebAppKind != \"\""
+            "visibleRule": "ConnectionType = AzureRM || WebAppKind != \"\""
```

Comment:
You should also have the connectionType condition in AdditionalDeploymentOptions group

Reply:
Done.